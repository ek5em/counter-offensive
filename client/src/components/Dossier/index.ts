import Info from "./Info/Info";
import Shop from "./Shop/Shop";

export { Info, Shop };
