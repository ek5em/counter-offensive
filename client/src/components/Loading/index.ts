import CardAdvise from "./CardAdvice/CardAdvice";
import ProgressBar from "./ProgressBar/ProgressBar";

export { ProgressBar, CardAdvise };
