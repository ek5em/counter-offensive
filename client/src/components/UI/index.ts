import Button from "./Button/Button";
import Eye from "./Eye/Eye";
import Input from "./Input/Input";
import Logo from "./Logo/Logo";
import { Alert } from './Alert/Alert';
export { Logo, Input, Button, Eye, Alert };
