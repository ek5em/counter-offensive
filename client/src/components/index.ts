import Login from "./Login/Login";
import Registration from "./Registration/Registration";

export { Shop, Info } from "./Dossier";
export { ProgressBar, CardAdvise } from "./Loading";
export { Logo, Button, Input, Eye } from "./UI";
export { Login };
export { Registration };