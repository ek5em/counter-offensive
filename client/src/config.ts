// dev

/* const PORT = 81;
const DOMAIN = 'http://counter-offensive'; */

//prod

const PORT = 3000;
const DOMAIN = "http://localhost";

export const HOST = PORT ? `${DOMAIN}:${PORT}` : DOMAIN;

export const MEDIATOR = {
    EVENTS: {
        SERVER_ERROR: "SERVER_ERROR",
        GO_TO_TANK: "GO_TO_TANK",
        UPDATE_USER: "UPDATE_USER",
        UPDATE_TOKEN: "UPDATE_TOKEN",
        LOBBY_UPDATE: "LOBBY_UPDATE",
        END_GAME: "END_GAME",
        WARNING: "WARNING",
        AUTH_ERROR: "AUTH_ERROR",
        ROLE_ERROR: "ROLE_ERROR",
        THROW_TO_GAME: "THROW_TO_GAME",
        THROW_TO_LOBBY: "THROW_TO_LOBBY",
        UPDATE_SCENE: "UPDATE_SCENE",
        UPDATE_TIME: "UPDATE_TIME",
        NEW_MESSAGE: "NEW_MESSAGE",
        SEND_MESSAGE_STATUS: "SEND_MESSAGE_STATUS",
        MOVE_UNIT: "MOVE_UNIT",
    },

    TRIGGERS: {
        LOBBY: "LOBBY",
        TOKEN: "TOKEN",
        USER: "USER",
    },
};

export enum ESOCKET {
    SEND_MESSAGE = "SEND_MESSAGE",
    GET_MESSAGE = "GET_MESSAGE",
    REGISTRATION = "REGISTRATION",
    LOGIN = "LOGIN",
    LOGOUT = "LOGOUT",
    UPDATE_PASSWORD = "UPDATE_PASSWORD",
    SET_GAMER_ROLE = "SET_GAMER_ROLE",
    GET_LOBBY = "GET_LOBBY",
    SUICIDE = "SUICIDE",
    GET_USER_INFO = "GET_USER_INFO",
    GET_SCENE = "GET_SCENE",
    MOTION = "MOTION",
    FIRE = "FIRE",
    ERROR = "ERROR",
    TOKEN_VERIFICATION = "TOKEN_VERIFICATION",
    GAME_MAP = "GAME_MAP",
    GAME_ENTITIES = "GAME_ENTITIES",
}

export const MAP_SIZE = {
    width: 150,
    height: 120,
};

export const entitiesConfig = {
    infantry: {
        speed: 0.002,
        r: 0.35,
        weaponLength: 0.6,
        visiableAngle: 120,
    },

    infantryRGP: {
        speed: 0.0018,
        r: 0.35,
        weaponLength: 0.6,
        visiableAngle: 120,
    },

    middleTank: {
        weaponLength: 1,
        speed: 0.005,
        corpusR: 1,
        towerR: 0,
        rotateSpeed: (Math.PI / 180) * 0.04,
        rotateTowerSpeed: (Math.PI / 180) * 0.04,
        visiableAngle: {
            gunner: 45,
            driver: 60,
        },
    },

    heavyTank: {
        weaponLength: 1,
        speed: 0.004,
        corpusR: 1.2,
        towerR: 0,
        rotateSpeed: (Math.PI / 180) * 0.035,
        rotateTowerSpeed: (Math.PI / 180) * 0.04,
        visiableAngle: {
            gunner: 45,
            driver: 60,
            comander: 180,
        },
    },

    bannerman: {
        speed: 0.05 /* 0.0015 */,
        r: 0.35,
        visiableAngle: 120,
    },

    general: {
        speed: 0.06,
    },

    bulletSpeed: 0.1,
};

export const objectConf = {
    stone: {
        r: 1,
    },
    house: {
        width: 6,
        height: 3,
    },
    stump: {
        r: 1,
    },
    bush: {
        r: 1,
    },
    tree: {
        r: 1.5,
    },
    base: {
        r: 2,
        innerColor: "#8BD033",
        outerColor: "#FF0000",
    },
};

export const WINConf = {
    width: 40,
    height: 40,
};

export const walls = [
    [
        { x: -1, y: -1 },
        { x: -1, y: MAP_SIZE.height + 1 },
        { x: 0, y: MAP_SIZE.height + 1 },
        { x: 0, y: -1 },
    ],
    [
        { x: -1, y: -1 },
        { x: -1, y: 0 },
        { x: MAP_SIZE.width, y: 0 },
        { x: MAP_SIZE.width, y: -1 },
    ],
    [
        { x: MAP_SIZE.width, y: -1 },
        { x: MAP_SIZE.width, y: MAP_SIZE.height + 1 },
        { x: MAP_SIZE.width + 1, y: MAP_SIZE.height + 1 },
        { x: MAP_SIZE.width + 1, y: -1 },
    ],
    [
        { x: 0, y: MAP_SIZE.height },
        { x: 0, y: MAP_SIZE.height + 1 },
        { x: MAP_SIZE.width, y: MAP_SIZE.height + 1 },
        { x: MAP_SIZE.width, y: MAP_SIZE.height },
    ],
];

export const staticMap = {
    bushes: [
        { x: 4, y: 28 },
        { x: 2, y: 32 },
        { x: 5, y: 33 },
        { x: 3, y: 43 },
        { x: 6, y: 44 },
        { x: 12, y: 46 },
        { x: 24, y: 46 },
        { x: 6, y: 54 },
        { x: 9, y: 55 },
        { x: 10, y: 98 },
        { x: 11, y: 95 },
        { x: 8, y: 112 },
        { x: 42, y: 2 },
        { x: 45, y: 3 },
        { x: 44, y: 8 },
        { x: 47, y: 13 },
        { x: 60, y: 13 },
        { x: 32, y: 14 },
        { x: 36, y: 22 },
        { x: 37, y: 27 },
        { x: 45, y: 31 },
        { x: 39, y: 32 },
        { x: 28, y: 28 },
        { x: 24, y: 30 },
        { x: 27, y: 37 },
        { x: 51, y: 41 },
        { x: 50, y: 44 },
        { x: 45, y: 43 },
        { x: 48, y: 50 },
        { x: 64, y: 50 },
        { x: 49, y: 59 },
        { x: 52, y: 60 },
        { x: 57, y: 61 },
        { x: 56, y: 64 },
        { x: 56, y: 64 },
        { x: 63, y: 77 },
        { x: 66, y: 78 },
        { x: 65, y: 75 },
        { x: 64, y: 78 },
        { x: 22, y: 68 },
        { x: 21, y: 71 },
        { x: 24, y: 95 },
        { x: 20, y: 100 },
        { x: 32, y: 118 },
        { x: 38, y: 117 },
        { x: 34, y: 111 },
        { x: 24, y: 111 },
        { x: 36, y: 106 },
        { x: 66, y: 26 },
        { x: 62, y: 27 },
        { x: 63, y: 31 },
        { x: 76, y: 22 },
        { x: 79, y: 32 },
        { x: 75, y: 74 },
        { x: 80, y: 77 },
        { x: 73, y: 91 },
        { x: 77, y: 92 },
        { x: 84, y: 14 },
        { x: 81, y: 101 },
        { x: 85, y: 101 },
        { x: 89, y: 101 },
        { x: 80, y: 105 },
        { x: 84, y: 105 },
        { x: 88, y: 105 },
        { x: 81, y: 109 },
        { x: 85, y: 109 },
        { x: 89, y: 109 },
        { x: 80, y: 113 },
        { x: 84, y: 113 },
        { x: 88, y: 113 },
        { x: 94, y: 8 },
        { x: 91, y: 15 },
        { x: 94, y: 23 },
        { x: 97, y: 41 },
        { x: 90, y: 39 },
        { x: 91, y: 57 },
        { x: 95, y: 58 },
        { x: 93, y: 60 },
        { x: 90, y: 65 },
        { x: 92, y: 68 },
        { x: 96, y: 66 },
        { x: 93, y: 73 },
        { x: 92, y: 76 },
        { x: 96, y: 76 },
        { x: 96, y: 100 },
        { x: 95, y: 109 },
        { x: 98, y: 110 },
        { x: 101, y: 25 },
        { x: 117, y: 64 },
        { x: 114, y: 65 },
        { x: 116, y: 70 },
        { x: 117, y: 81 },
        { x: 114, y: 83 },
        { x: 116, y: 85 },
        { x: 114, y: 112 },
        { x: 113, y: 115 },
        { x: 126, y: 8 },
        { x: 121, y: 12 },
        { x: 121, y: 12 },
        { x: 126, y: 31 },
        { x: 125, y: 32 },
        { x: 124, y: 33 },
        { x: 125, y: 49 },
        { x: 130, y: 50 },
        { x: 125, y: 65 },
        { x: 122, y: 67 },
        { x: 124, y: 69 },
        { x: 128, y: 114 },
        { x: 127, y: 117 },
        { x: 135, y: 8 },
        { x: 131, y: 21 },
        { x: 133, y: 41 },
        { x: 136, y: 42 },
        { x: 132, y: 44 },
        { x: 133, y: 49 },
        { x: 138, y: 52 },
        { x: 137, y: 55 },
        { x: 139, y: 79 },
        { x: 138, y: 80 },
        { x: 137, y: 81 },
        { x: 136, y: 86 },
        { x: 137, y: 87 },
        { x: 136, y: 88 },
        { x: 135, y: 95 },
        { x: 138, y: 96 },
        { x: 134, y: 98 },
        { x: 132, y: 116 },
        { x: 141, y: 5 },
        { x: 148, y: 6 },
        { x: 144, y: 20 },
        { x: 143, y: 22 },
        { x: 146, y: 22 },
        { x: 141, y: 55 },
        { x: 146, y: 66 },
        { x: 142, y: 69 },
        { x: 146, y: 70 },
    ],
    trees: [
        { x: 12, y: 79 },
        { x: 35, y: 87 },
        { x: 31, y: 109 },
        { x: 48, y: 29 },
        { x: 41, y: 76 },
        { x: 44, y: 79 },
        { x: 42, y: 110 },
        { x: 53, y: 79 },
        { x: 56, y: 99 },
        { x: 69, y: 51 },
        { x: 72, y: 7 },
        { x: 76, y: 59 },
        { x: 72, y: 99 },
        { x: 87, y: 5 },
        { x: 82, y: 36 },
        { x: 82, y: 45 },
        { x: 82, y: 45 },
        { x: 98, y: 31 },
        { x: 98, y: 63 },
        { x: 96, y: 117 },
        { x: 103, y: 67 },
        { x: 116, y: 10 },
        { x: 118, y: 21 },
        { x: 124, y: 14 },
        { x: 124, y: 24 },
        { x: 130, y: 7 },
        { x: 136, y: 15 },
        { x: 135, y: 21 },
    ],
    roads: [{ x: 108, y: 0, sizeX: 3, sizeY: 120 }],
    crossyRoads: [
        {
            x: 16,
            y: 18,
            sizeX: 2,
            sizeY: 10,
            angle: Math.PI / 2,
        },
        {
            x: 18,
            y: 24,
            sizeX: 14,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 32,
            y: 26,
            sizeX: 2,
            sizeY: 62,
            angle: 90,
        },
        {
            x: 18,
            y: 64,
            sizeX: 14,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 14,
            y: 66,
            sizeX: 2,
            sizeY: 6,
            angle: 90,
        },
        {
            x: 18,
            y: 84,
            sizeX: 14,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 16,
            y: 86,
            sizeX: 2,
            sizeY: 16,
            angle: 90,
        },
        {
            x: 18,
            y: 96,
            sizeX: 6,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 24,
            y: 98,
            sizeX: 2,
            sizeY: 2,
            angle: 90,
        },
        {
            x: 34,
            y: 88,
            sizeX: 74,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 60,
            y: 90,
            sizeX: 2,
            sizeY: 8,
            angle: 90,
        },
        {
            x: 62,
            y: 98,
            sizeX: 4,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 66,
            y: 100,
            sizeX: 2,
            sizeY: 4,
            angle: 90,
        },
        {
            x: 16,
            y: 62,
            sizeX: 2,
            sizeY: 2,
            angle: 90,
        },
        {
            x: 72,
            y: 54,
            sizeX: 2,
            sizeY: 34,
            angle: 90,
        },
        {
            x: 74,
            y: 68,
            sizeX: 2,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 74,
            y: 78,
            sizeX: 34,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 34,
            y: 52,
            sizeX: 74,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 54,
            y: 82,
            sizeX: 2,
            sizeY: 6,
            angle: 90,
        },
        {
            x: 40,
            y: 80,
            sizeX: 14,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 40,
            y: 70,
            sizeX: 8,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 48,
            y: 72,
            sizeX: 2,
            sizeY: 8,
            angle: 90,
        },
        {
            x: 38,
            y: 60,
            sizeX: 2,
            sizeY: 20,
            angle: 90,
        },
        {
            x: 42,
            y: 54,
            sizeX: 2,
            sizeY: 4,
            angle: 90,
        },
        {
            x: 40,
            y: 58,
            sizeX: 2,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 66,
            y: 48,
            sizeX: 2,
            sizeY: 4,
            angle: 90,
        },
        {
            x: 84,
            y: 48,
            sizeX: 2,
            sizeY: 4,
            angle: 90,
        },
        {
            x: 68,
            y: 46,
            sizeX: 16,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 18,
            y: 16,
            sizeX: 90,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 40,
            y: 8,
            sizeX: 2,
            sizeY: 8,
            angle: 90,
        },
        {
            x: 48,
            y: 18,
            sizeX: 2,
            sizeY: 2,
            angle: 90,
        },
        {
            x: 76,
            y: 10,
            sizeX: 2,
            sizeY: 6,
            angle: 90,
        },
        {
            x: 78,
            y: 18,
            sizeX: 2,
            sizeY: 6,
            angle: 90,
        },
        {
            x: 110,
            y: 16,
            sizeX: 40,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 110,
            y: 44,
            sizeX: 18,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 126,
            y: 40,
            sizeX: 2,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 128,
            y: 38,
            sizeX: 2,
            sizeY: 6,
            angle: 90,
        },
        {
            x: 130,
            y: 36,
            sizeX: 8,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 110,
            y: 52,
            sizeX: 18,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 128,
            y: 54,
            sizeX: 2,
            sizeY: 6,
            angle: 90,
        },
        {
            x: 110,
            y: 76,
            sizeX: 32,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 128,
            y: 78,
            sizeX: 2,
            sizeY: 2,
            angle: 90,
        },
        {
            x: 142,
            y: 78,
            sizeX: 2,
            sizeY: 2,
            angle: 90,
        },
        {
            x: 110,
            y: 104,
            sizeX: 26,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 140,
            y: 102,
            sizeX: 10,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 138,
            y: 110,
            sizeX: 2,
            sizeY: 2,
            angle: 0,
        },
        {
            x: 120,
            y: 106,
            sizeX: 2,
            sizeY: 2,
            angle: 90,
        },
        {
            x: 122,
            y: 102,
            sizeX: 2,
            sizeY: 2,
            angle: 90,
        },
        {
            x: 138,
            y: 100,
            sizeX: 2,
            sizeY: 2,
            angle: 90,
        },
        {
            x: 136,
            y: 106,
            sizeX: 2,
            sizeY: 4,
            angle: 90,
        },
    ],
    crossyRoadsTurnCont: [
        { x: 42, y: 52, angle: 0 },
        { x: 60, y: 88, angle: 0 },
        { x: 48, y: 16, angle: 0 },
        { x: 78, y: 16, angle: 0 },
        { x: 72, y: 52, angle: 0 },
        { x: 120, y: 104, angle: 0 },
        { x: 136, y: 104, angle: 0 },
        { x: 128, y: 76, angle: 0 },

        { x: 16, y: 96, angle: 90 },
        { x: 16, y: 24, angle: 90 },
        { x: 38, y: 70, angle: 90 },
        { x: 32, y: 46, angle: 90 },
        { x: 32, y: 52, angle: 90 },
        { x: 66, y: 46, angle: 90 },
        { x: 72, y: 68, angle: 90 },
        { x: 72, y: 78, angle: 90 },
        { x: 138, y: 102, angle: 90 },

        { x: 16, y: 64, angle: 180 },
        { x: 40, y: 16, angle: 180 },
        { x: 48, y: 80, angle: 180 },
        { x: 54, y: 88, angle: 180 },
        { x: 72, y: 88, angle: 180 },
        { x: 84, y: 52, angle: 180 },
        { x: 76, y: 16, angle: 180 },
        { x: 66, y: 52, angle: 180 },
        { x: 134, y: 36, angle: 180 },
        { x: 122, y: 104, angle: 180 },

        { x: 32, y: 64, angle: 270 },
        { x: 32, y: 84, angle: 270 },
        { x: 128, y: 40, angle: 270 },
    ],
    crossyRoadsTurn: [
        { x: 32, y: 24, angle: 0 },
        { x: 24, y: 96, angle: 0 },
        { x: 48, y: 70, angle: 0 },
        { x: 54, y: 80, angle: 0 },
        { x: 66, y: 98, angle: 0 },
        { x: 84, y: 46, angle: 0 },
        { x: 142, y: 76, angle: 0 },
        { x: 128, y: 52, angle: 0 },

        { x: 16, y: 102, angle: 90 },
        { x: 42, y: 58, angle: 90 },
        { x: 78, y: 24, angle: 90 },
        { x: 128, y: 44, angle: 90 },
        { x: 138, y: 104, angle: 90 },

        { x: 32, y: 88, angle: 180 },
        { x: 38, y: 80, angle: 180 },
        { x: 60, y: 98, angle: 180 },
        { x: 136, y: 110, angle: 180 },

        { x: 14, y: 64, angle: 270 },
        { x: 16, y: 16, angle: 270 },
        { x: 16, y: 84, angle: 270 },
        { x: 38, y: 58, angle: 270 },
        { x: 76, y: 8, angle: 270 },
        { x: 128, y: 36, angle: 270 },
    ],
    crossyRoadsEnd: [
        { x: 14, y: 102, angle: 0 },
        { x: 76, y: 24, angle: 0 },
        { x: 124, y: 40, angle: 0 },

        { x: 16, y: 60, angle: 90 },
        { x: 40, y: 6, angle: 90 },
        { x: 66, y: 44, angle: 90 },
        { x: 122, y: 100, angle: 90 },
        { x: 138, y: 98, angle: 90 },
        { x: 134, y: 34, angle: 90 },

        { x: 34, y: 46, angle: 180 },
        { x: 34, y: 46, angle: 180 },
        { x: 78, y: 8, angle: 180 },
        { x: 76, y: 68, angle: 180 },
        { x: 140, y: 110, angle: 180 },
        { x: 138, y: 36, angle: 180 },

        { x: 16, y: 28, angle: 270 },
        { x: 14, y: 72, angle: 270 },
        { x: 24, y: 100, angle: 270 },
        { x: 48, y: 20, angle: 270 },
        { x: 66, y: 104, angle: 270 },
        { x: 128, y: 60, angle: 270 },
        { x: 128, y: 80, angle: 270 },
        { x: 142, y: 80, angle: 270 },
        { x: 120, y: 108, angle: 270 },
    ],
};
