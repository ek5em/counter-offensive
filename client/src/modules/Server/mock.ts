import {
    EGamerRole,
    ERank,
    IGamerInfo,
    ILobby,
    IMessage,
    IUserUnit,
} from "./interfaces";

interface IMock {
    users: (IGamerInfo & { password: string })[];
    messages: {
        text: string;
        rank_name: ERank;
        userId: number;
        nickname: string;
    }[];
    lobby: ILobby;
    unit: IUserUnit;
}

export const MOCK: IMock = {
    users: [
        {
            gamer_exp: 9999,
            id: 0,
            is_alive: null,
            level: 10,
            login: "123",
            next_rank: 65421,
            nickname: "Гришка",
            password: "123",
            rank_name: ERank.General,
            token: "123",
        },
        {
            gamer_exp: 9999,
            id: 1,
            is_alive: null,
            level: 10,
            login: "321",
            next_rank: 65421,
            nickname: "Петька заводной",
            password: "321",
            rank_name: ERank.General,
            token: "321",
        },
    ],
    messages: [
        {
            nickname: "Гришка",
            rank_name: ERank.General,
            text: "Привет, парни!",
            userId: 0,
        },
        {
            nickname: "Петька заводной",
            rank_name: ERank.General,
            text: "Привет, Гришка, ну чо как вчера жёско зарубимся в контр-нахрюк???!",
            userId: 1,
        },
    ],
    lobby: {
        bannerman: true,
        general: true,
        tanks: {
            heavyTank: [
                { id: 1, Commander: true, Gunner: false, Mechanic: false },
                { id: 3, Commander: false, Gunner: true, Mechanic: false },
            ],
            middleTank: [{ id: 2, Gunner: true, Mechanic: false }],
        },
    },
    unit: {
        angle: 0,
        personId: EGamerRole.infantry,
        speed: 1,
        x: 5,
        y: 5,
    },
};
