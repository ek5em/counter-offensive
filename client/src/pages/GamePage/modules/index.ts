import Collision from "./Game/Collision/Collision";
import MathGame from "./Game/Math/MathGame";

import Game from "./Game/Game";
import Canvas from "./Graph/Canvas/Canvas";
import TraceMask from "./Graph/TraceMask";

export { Canvas, Collision, Game, TraceMask, MathGame };

export * from "./Game/Units";
