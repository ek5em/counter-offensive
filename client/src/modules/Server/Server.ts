import { SHA256 } from "crypto-js";
import { io, Socket } from "socket.io-client";
import Mediator from "../Mediator/Mediator";
import Store from "../Store/Store";
import {
    IGamerInfo,
    IError,
    EGamerRole,
    IScene,
    IToken,
    IAnswer,
    ILobby,
    ETank,
    IMessage,
} from "./interfaces";
import { ESOCKET } from "../../config";

export default class Server {
    mediator: Mediator;
    HOST: string;
    STORE: Store;
    socket: Socket;

    constructor(HOST: string, mediator: Mediator) {
        this.HOST = HOST;
        this.mediator = mediator;
        this.STORE = new Store();

        this.socket = io(HOST);

        const {
            NEW_MESSAGE,
            LOGIN,
            SEND_MESSAGE_STATUS,
            LOGOUT,
            UPDATE_SCENE,
        } = mediator.getTriggerTypes();

        const { SERVER_ERROR, GO_TO_TANK, UPDATE_USER, LOBBY_UPDATE } =
            this.mediator.getEventTypes();

        this.socket.on(ESOCKET.ERROR, (answer: IError) => {
            mediator.call(SERVER_ERROR, answer.error);
        });

        this.socket.on(ESOCKET.GET_MESSAGE, (answer: IAnswer<IMessage[]>) => {
            mediator.get(NEW_MESSAGE, answer.data);
        });

        this.socket.on(ESOCKET.SEND_MESSAGE, (answer: IAnswer<true>) => {
            mediator.get(SEND_MESSAGE_STATUS, answer.data);
        });

        this.socket.on(ESOCKET.REGISTRATION, (answer: IAnswer<IToken>) => {
            mediator.get(LOGIN, answer.data);
        });

        this.socket.on(ESOCKET.LOGIN, (answer: IAnswer<IToken>) => {
            mediator.get(LOGIN, answer.data);
        });

        this.socket.on(ESOCKET.GET_USER_INFO, (answer: IAnswer<IGamerInfo>) => {
            this.STORE.setUser(answer.data);
            mediator.call(UPDATE_USER, answer.data);
        });

        this.socket.on(ESOCKET.LOGOUT, () => {
            mediator.get(LOGOUT);
        });

        this.socket.on(ESOCKET.TOKEN_VERIFICATION, () => {
            mediator.get(LOGIN, { token: this.STORE.getToken() });
        });

        this.socket.on(
            ESOCKET.SET_GAMER_ROLE,
            (answer: IAnswer<{ tankId: number; tankType: ETank }>) => {
                if (answer.data.tankId) {
                    mediator.call(GO_TO_TANK, answer.data);
                }
            }
        );

        this.socket.on(ESOCKET.GET_LOBBY, (answer: IAnswer<ILobby>) => {
            this.STORE.setLobby(answer.data);
            mediator.call(LOBBY_UPDATE, answer.data);
        });

        this.socket.on(ESOCKET.GET_SCENE, (answer: IAnswer<IScene>) => {
            mediator.get(UPDATE_SCENE, answer.data);
        });
    }

    // АВТОРИЗАЦИЯ

    registration(login: string, nickname: string, password: string) {
        const hash = SHA256(login + password).toString();
        this.socket.emit(ESOCKET.REGISTRATION, { login, nickname, hash });
    }

    login(login: string, password: string) {
        const rnd = Math.random();
        const hash = SHA256(
            SHA256(login + password).toString() + rnd
        ).toString();
        this.socket.emit(ESOCKET.LOGIN, { login, hash, rnd });
    }

    tokenVerification() {
        this.socket.emit(ESOCKET.TOKEN_VERIFICATION, {
            token: this.STORE.getToken(),
        });
    }

    getUser() {
        this.socket.emit(ESOCKET.GET_USER_INFO, {
            token: this.STORE.getToken(),
        });
    }

    logout() {
        this.socket.emit(ESOCKET.LOGOUT, {
            token: this.STORE.getToken(),
        });
    }

    // ЛОББИ

    sendMessage(message: string) {
        this.socket.emit(ESOCKET.SEND_MESSAGE, {
            token: this.STORE.getToken(),
            message,
        });
    }

    getMessages() {
        this.socket.emit(ESOCKET.GET_MESSAGE, {
            token: this.STORE.getToken(),
        });
    }

    setGamerRole(role: EGamerRole, tankId: number | null = null) {
        this.socket.emit(ESOCKET.SET_GAMER_ROLE, {
            token: this.STORE.getToken(),
            role,
            tankId,
        });
    }

    getLobby() {
        this.socket.emit(ESOCKET.GET_LOBBY, {
            token: this.STORE.getToken(),
        });
    }

    // ИГРА

    getScene() {
        this.socket.emit(ESOCKET.GET_SCENE, {
            token: this.STORE.getToken(),
        });
    }

    suicide() {
        this.socket.emit(ESOCKET.SUICIDE, {
            token: this.STORE.getToken(),
        });
    }

    unitMotion(x: number | null, y: number | null, angle: number) {
        this.socket.emit(ESOCKET.MOTION, {
            token: this.STORE.getToken(),
            x,
            y,
            angle,
        });
    }

    makeShot(x: number, y: number, angle: number) {
        this.socket.emit(ESOCKET.FIRE, {
            token: this.STORE.getToken(),
            x,
            y,
            angle,
        });
    }
}
