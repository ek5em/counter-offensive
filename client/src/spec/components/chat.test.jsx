import { useState } from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Chat, ContextProvider, EChat } from "../../components";
import { MOCK } from "../../modules/Server/mock";
import { Mediator, Server } from "../../modules";
import { MEDIATOR, HOST } from "../../config";
import { MediatorContext, ServerContext } from "../../components";

window.HTMLElement.prototype.scrollIntoView = function () {};

describe("Chat component", () => {
    const ChatTestWrapper = ({ initialIsOpen }) => {
        const [isOpen, setIsOpen] = useState(initialIsOpen);
        return (
            <Chat
                chatType={EChat.lobby}
                isOpen={isOpen}
                setIsOpen={() => setIsOpen(!isOpen)}
            />
        );
    };

    test("Открытие чата", () => {
        render(
            <ContextProvider>
                <ChatTestWrapper initialIsOpen={false} />
            </ContextProvider>
        );
        const openButton = screen.getByTestId("chat-opener");
        expect(openButton).toBeInTheDocument();
        fireEvent.click(openButton);
        const closeButton = screen.getByTestId("chat-closer");
        expect(closeButton).toBeInTheDocument();
        expect(screen.getByTestId("chat-input")).toBeInTheDocument();
    });

    test("Закрытие чата", () => {
        render(
            <ContextProvider>
                <ChatTestWrapper initialIsOpen={true} />
            </ContextProvider>
        );
        const closeButton = screen.getByTestId("chat-closer");
        expect(closeButton).toBeInTheDocument();
        fireEvent.click(closeButton);
        const openButton = screen.getByTestId("chat-opener");
        expect(openButton).toBeInTheDocument();
    });

    test("Получение сообщений", () => {
        render(
            <ContextProvider useMock={true}>
                <ChatTestWrapper initialIsOpen={true} />
            </ContextProvider>
        );
        MOCK.messages.forEach((message) => {
            expect(screen.getByText(message.text)).toBeInTheDocument();
        });
    });

    test("Отправка сообщений", () => {
        const newMessage = "1233221";
        const mediator = new Mediator(MEDIATOR);
        const server = new Server(HOST, mediator, true);
        const component = render(
            <BrowserRouter>
                <MediatorContext.Provider value={mediator}>
                    <ServerContext.Provider value={server}>
                        <ChatTestWrapper initialIsOpen={true} />
                    </ServerContext.Provider>
                </MediatorContext.Provider>
            </BrowserRouter>
        );
        //авторизация
        const { login, password } = MOCK.users[0];
        server.login(login, password);
        const input = screen.getByPlaceholderText("Написать в чат");
        fireEvent.change(input, { target: { value: newMessage } });
        // Проверяем, что текст был успешно введен в инпут
        expect(input.value).toBe(newMessage);
        const button = screen.getByRole("button", { name: /send/i });
        fireEvent.click(button);
        expect(input.value).toBe("");
        expect(screen.getByText(newMessage));
        expect(component.asFragment()).toMatchSnapshot();
    });
});
