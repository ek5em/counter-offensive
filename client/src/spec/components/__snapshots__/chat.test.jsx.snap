// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component Отправка сообщений 1`] = `
<DocumentFragment>
  <div
    class="lobby chat"
  >
    <div
      class="title"
    >
      <span>
        Чат
      </span>
      <img
        class="close"
        data-testid="chat-closer"
        src="closeIcon.png"
      />
    </div>
    <div>
      <div
        class="messages"
      >
        <div
          class="message"
        >
          <span
            class="name"
          >
            [Гришка
            <img
              alt="rank"
              class="rank_img"
              src="fourthRank.png"
            />
            ]
          </span>
          : 
          <span
            class="message_text"
          >
            1233221
          </span>
        </div>
        <div
          class="message"
        >
          <span
            class="name"
          >
            [Петька заводной
            <img
              alt="rank"
              class="rank_img"
              src="fourthRank.png"
            />
            ]
          </span>
          : 
          <span
            class="message_text"
          >
            Привет, Гришка, ну чо как вчера жёско зарубимся в контр-нахрюк???!
          </span>
        </div>
        <div
          class="message"
        >
          <span
            class="name"
          >
            [Гришка
            <img
              alt="rank"
              class="rank_img"
              src="fourthRank.png"
            />
            ]
          </span>
          : 
          <span
            class="message_text"
          >
            Привет, парни!
          </span>
        </div>
        <div />
      </div>
      <form
        class="form"
      >
        <input
          class="input_chat"
          data-testid="chat-input"
          placeholder="Написать в чат"
          type="text"
          value=""
        />
        <button
          class="send"
        >
          <img
            alt="Send"
            src="sendMessage.png"
          />
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
